components/scream/src/physics/p3/scream_p3_interface.F90

```diff
@@ -57,26 +57,27 @@ subroutine p3_init_f90 () bind(c)
 
   end subroutine p3_init_f90
   !====================================================================!
-  subroutine p3_standalone_init_f90 (q,T,zi,pmid,pdel,ast,naai,npccn) bind(c)
+  subroutine p3_standalone_init_f90 (qdp,zi,pmid,pdel,ast,naai,npccn) bind(c)
     use micro_p3,       only: p3_init
     use micro_p3_utils, only: micro_p3_utils_init
 
-    real(kind=c_real), intent(inout) :: q(pcols,pver,9)      ! State array  kg/kg
-    real(kind=c_real), intent(inout) :: T(pcols,pver)        ! 
+    real(kind=c_real), intent(inout) :: qdp(pcols,pver,qsize)    ! State array  kg/kg Pa
     real(kind=c_real), intent(inout) :: zi(pcols,pver+1)     ! 
     real(kind=c_real), intent(inout) :: pmid(pcols,pver)     ! 
     real(kind=c_real), intent(inout) :: pdel(pcols,pver)     ! 
     real(kind=c_real), intent(inout) :: ast(pcols,pver)      ! 
     real(kind=c_real), intent(inout) :: naai(pcols,pver)     ! ice nucleation number
     real(kind=c_real), intent(inout) :: npccn(pcols,pver)    ! liquid activation number tendency
 
+    real(kind=c_real) :: T(pcols,pver)
+
     character(len=100) :: case_title
 
     integer(kind=c_int) :: i, k
     logical(kind=c_bool) :: masterproc
 
     ! READ inputs from SCM for p3-stand-alone:
-    q(:,:,:) = 0.0_rtype
+    qdp(:,:,:) = 0.0_rtype
```

Comment:
I believe the presence of `qdp` in the original p3_standalone code confused things.  The original intent was just to check that a variable from dynamics was making it into physics.  But P3 itself does not use qdp, it just uses q.  We can remove qdp and any reference to it entirely from the P3 code and revert all the `q` references back to just plain `q`.

Reply:
Ok, I'm going to purge qdp completely.

Reply:
Done!